### PR TITLE
Initial helm chart setup for hedera json-rpc relay. 

### LINKED ISSUE #61: Add simple Helm chart to support deployment
Chart should utilize deployed docker image and support simple k8s deployment for varied environments

Can draw inspiration from [Mirror Node charts](https://github.com/hashgraph/hedera-mirror-node/tree/main/charts)
`hedera-mirror-web3` should be exactly what we need to start off with.
`hedera-mirror-rest` may show more mature logic and tie closely since it's also a node app

### PR DESCRIPTION
**Pull request by rustyShacklefurd**
**Description**:

**Related issue(s)**:

Fixes #61 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,6 +84,23 @@ curl -X POST -H 'Content-Type: application/json' -d '{"jsonrpc":"2.0","id":"2","
 The expected response should be `{"result":"0x12a","jsonrpc":"2.0","id":"2"}`
 Where the `result` value matches the .env `CHAIN_ID` configuration value or the current deault value of `298`
 
+### Helm Chart
+
+In this repo there is a `helm-chart` directory that contains the configurations for deploy Hedera's json-rpc relay to a K8s cluster.
```
Comment (Nana-EC) [inline README.md:89]: nit: 
```suggestion
In this repo there is a `helm-chart` directory that contains the configurations to deploy Hedera's json-rpc relay to a K8s cluster.
```

File: helm-chart/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v2
+name: hedera-json-rpc-relay
+description: Helm chart deployment of the hashgraph/hedera-json-rpc-relay
+type: application
+version: 0.0.1
+appVersion: "0.0.1"
+home: https://github.com/hashgraph/hedera-json-rpc-relay
```
Comment (Nana-EC) [inline helm-chart/Chart.yaml:7]: nit: add an extra line to all new files to git picks up commit changes more easily.
I also forgot this on one of my more recent PRs so i get it.

File: helm-chart/values.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+# Default values for json-rpc-relay.
+
+replicaCount: 1
+
+image:
+  repository: #IMAGE is not published yet, use internal repo/location to push to
```
Comment (Nana-EC) [inline helm-chart/values.yaml:6]: It's published, you can pull 
```suggestion
  repository: ghcr.io/hashgraph/hedera-json-rpc-relay:main
```

File: README.md [hunk 4]
```diff
@@ -84,6 +84,23 @@ curl -X POST -H 'Content-Type: application/json' -d '{"jsonrpc":"2.0","id":"2","
...
+To get started install the helm chart:
+```
+helm install hedera-json-rpc-relay ./helm-chart --debug
+```
+
+To see the values that have been deployed:
+```
+helm show values hedera-json-rpc-relay
+```
+Deploy an installation with custom values file:
```
Comment (Nana-EC) [inline README.md:99]: Can you provide an example of a custom file that lets you specify the current values in the `.env` file
This is how people would customize which env and which account the relay uses.

See the `### Configuration` section as an example above

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looks good, some minor requests

File: helm-chart/templates/service.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.ports.containerPort }}
+      targetPort: {{ .Values.ports.name  }}
+      protocol: TCP
+      name: {{ .Values.ports.name  }}
+  selector:
+    app: {{ template "json-rpc-relay.name" . }}
+    {{- include "json-rpc-relay.selectorLabels" . | nindent 4 }}
```
Comment (mmrt) [inline helm-chart/templates/service.yaml:16]: newline

File: helm-chart/templates/serviceaccount.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+kind: ServiceAccount
+metadata:
+  name: {{ include "json-rpc-relay.serviceAccountName" . }}
+  labels:
+    {{- include "json-rpc-relay.labels" . | nindent 4 }}
+  {{- with .Values.serviceAccount.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+{{- end }}
```
Comment (mmrt) [inline helm-chart/templates/serviceaccount.yaml:12]: newline

**Review by mmrt — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (rustyShacklefurd) [reply README.md:99]: I've added secret with mount point to support the .env the same way it is described here.  Also added a section in the README about setting up locally on minikube environment.

Comment (Nana-EC) [reply README.md:99]: Lovely

File: README.md [hunk 7]
```diff
@@ -100,6 +100,27 @@ Deploy an installation with custom values file:
...
+  * Acquire PAT, username, and, (primary) email address from Github.
+  * Manually create a secret on kubernetes with the following command.  The $ must be replaced
+    ```
+    kubectl create secret docker-registry ghcr-registry-auth \
+    --docker-server=https://ghcr.io \
+    --docker-username=$GH_USERNAME \
+    --docker-password=$GH_PAT \
+    --docker-email=$GH_EMAIL
+    ```
+3. Deploy this helm-chart with the addtion environment/minikube.yaml file
```
Comment (Nana-EC) [inline README.md:115]: Can also make it a linkable file
```suggestion
3. Deploy this helm-chart with the additional [environment/minikube.yaml](environment/minikube.yaml) file
```

File: helm-chart/environments/minkube.yaml [hunk 8]
```diff
@@ -0,0 +1,2 @@
+imagePullSecrets: 
+  - name: ghcr-registry-auth
```
Comment (Nana-EC) [inline helm-chart/environments/minkube.yaml:2]: nit: add the extra line for git diff

File: helm-chart/templates/secret.yaml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  labels: {{ include "json-rpc-relay.labels" . | nindent 4 }}
+  name: {{ include "json-rpc-relay.fullname" . }}
+type: Opaque
+stringData:
+  .env: |-
+    {{- toYaml .Values.config | nindent 4 }}
```
Comment (Nana-EC) [inline helm-chart/templates/secret.yaml:9]: nit: extra line for diff

File: helm-chart/values.yaml [hunk 10]
```diff
@@ -82,4 +82,12 @@ affinity: {}
...
\ No newline at end of file
+
+config:
+  HEDERA_NETWORK=
+  OPERATOR_ID_MAIN=
+  OPERATOR_KEY_MAIN=
+  OPERATOR_ID_ETH_SENDRAWTRANSACTION=
+  OPERATOR_KEY_ETH_SENDRAWTRANSACTION=
+  CHAIN_ID=
+  MIRROR_NODE_URL=
```
Comment (Nana-EC) [inline helm-chart/values.yaml:93]: nit: extra line for diff

Comment (Nana-EC) [inline helm-chart/values.yaml:93]: Add `LOG_LEVEL`

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Some additional suggestions.
Can probably merge after this

File: README.md (hunk 7 above)
Comment (rustyShacklefurd) [reply README.md]: added

File: README.md [hunk 11]
```diff
@@ -112,7 +112,7 @@ helm install custom-hedera-json-rpc-relay -f path/to/values/file.yaml ./helm-cha
     --docker-password=$GH_PAT \
     --docker-email=$GH_EMAIL
     ```
-3. Deploy this helm-chart with the addtion environment/minikube.yaml file
+3. Deploy this helm-chart with the addtion [environment/minikube.yaml](environment/minikube.yaml) file
```
Comment (Nana-EC) [inline README.md:115]:
```suggestion
3. Deploy this helm-chart with the additional [environment/minikube.yaml](environment/minikube.yaml) file
```

**Review by Nana-EC — APPROVED** [review]
Missed a typo but good to merge

File: README.md (hunk 11 above)
Comment (rustyShacklefurd) [reply README.md]: using the commit suggested did not go too well here.  Its impossible to do that as a signed commit.  Whoops, added a commit lower to address this.

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request closed without merging**